src/network/networkd-nexthop.c

```diff
@@ -924,17 +924,8 @@ int link_drop_nexthops(Link *link, bool only_static) {
                 if (!nexthop_exists(nexthop))
                         continue;
 
-                if (nexthop->source == NETWORK_CONFIG_SOURCE_FOREIGN) {
-                        if (only_static)
-                                continue;
-
-                        /* Do not mark foreign nexthop when KeepConfiguration= is enabled. */
-                        if (link->network &&
-                            FLAGS_SET(link->network->keep_configuration, KEEP_CONFIGURATION_STATIC))
-                                continue;
-
-                } else if (nexthop->source != NETWORK_CONFIG_SOURCE_STATIC)
-                        continue; /* Ignore dynamically configurad nexthops. */
```

Comment:
Previously this didn't look at `nexthop->protocol` at all, now it does. Is protocol always `RTPROT_STATIC` here? Otherwise this is a behavior change, while the commit message says it's not supposed to be a behavior change.

Reply:
The nexthops statically configured by networkd has RTPROT_STATIC with NETWORK_CONFIG_SOURCE_STATIC.
When configured dynamically, it has RTPROT_RA or so with NETWORK_CONFIG_SOURCE_NDISC.
Currently, the kernel does not configure any nexthops implicitly.
So, the behavior may be changed only when a user manually configures nexthops.

Reply:
I will extend the commit message.

Reply:
Mentioned the behavior change. PTAL.